src/realm/object-store/impl/deep_change_checker.hpp

```diff
@@ -48,9 +48,8 @@ struct ListChangeInfo {
 };
 
 struct TransactionChangeInfo {
-    std::vector<ListChangeInfo> lists;
+    std::vector<ListChangeInfo> collections;
```

Comment:
Consider changing the type to be called "CollectionChangeInfo"

Reply:
Good call. I took a look at all of the places related to this which were using the word "list" and changed a bunch to "collection". The two remaining exceptions are ListNotifier, which runs into the problem that we already have something named CollectionNotifier (and more broadly the problem that SDKs consider Results a collection but core kinda doesn't), and the KVO stuff which on the core side could work with all collections, but is NSArray-specific in obj-c.